beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -193,18 +194,18 @@ func validateIndexInCommittee(ctx context.Context, s *stateTrie.BeaconState, a *
 }
 
 // Validates that the incoming aggregate attestation is in the desired time range.
-func validateAggregateAttTime(attSlot uint64, genesisTime uint64) error {
-	// in milliseconds
-	attTime := 1000 * (genesisTime + (attSlot * params.BeaconConfig().SecondsPerSlot))
+func validateAggregateAttTime(attSlot uint64, genesisTime time.Time) error {
+	// set expected boundaries for attestations
+	attTime := genesisTime.Add(time.Duration(attSlot*params.BeaconConfig().SecondsPerSlot) * time.Second)
 	attSlotRange := attSlot + params.BeaconNetworkConfig().AttestationPropagationSlotRange
-	attTimeRange := 1000 * (genesisTime + (attSlotRange * params.BeaconConfig().SecondsPerSlot))
-	currentTimeInSec := roughtime.Now().Unix()
-	currentTime := 1000 * currentTimeInSec
+	attTimeRange := genesisTime.Add(time.Duration(attSlotRange*params.BeaconConfig().SecondsPerSlot) * time.Second)
+	currentTime := roughtime.Now()
+	clockDisparity := params.BeaconNetworkConfig().MaximumGossipClockDisparity
 
 	// Verify attestation slot is within the last ATTESTATION_PROPAGATION_SLOT_RANGE slots.
-	currentSlot := (uint64(currentTimeInSec) - genesisTime) / params.BeaconConfig().SecondsPerSlot
-	if attTime-uint64(maximumGossipClockDisparity.Milliseconds()) > uint64(currentTime) ||
-		uint64(currentTime-maximumGossipClockDisparity.Milliseconds()) > attTimeRange {
+	currentSlot := helpers.SlotsSince(genesisTime)
+	if attTime.Add(-clockDisparity).After(currentTime) ||
+		currentTime.Add(-clockDisparity).After(attTimeRange) {
```

Comment:
Why not use `attTime.Sub(clockDisparity)` instead of `attTime.Add(-clockDisparity)` for the 2 lines above?

Reply:
In the official godoc of the `time` library. https://golang.org/pkg/time/#Time.Sub
```
// Sub returns the duration t-u. If the result exceeds the maximum (or minimum)
// value that can be stored in a Duration, the maximum (or minimum) duration
// will be returned.
// To compute t-d for a duration d, use t.Add(-d).
```
We would use `Sub` to compute the difference between two time instances. Whereas for the code, `clockDisparity` is a `time.Duration` value. So it made sense for me to use `Add` here instead.

Reply:
got it. thanks